tests/components/apns/test_notify.py

```diff
@@ -23,356 +23,367 @@
 }
 
 
-@patch("homeassistant.components.apns.notify.open", mock_open(), create=True)
-class TestApns(unittest.TestCase):
-    """Test the APNS component."""
-
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    @patch("os.path.isfile", Mock(return_value=True))
-    @patch("os.access", Mock(return_value=True))
-    def _setup_notify(self):
-        assert isinstance(apns.load_yaml_config_file, Mock), "Found unmocked load_yaml"
-
-        with assert_setup_component(1) as handle_config:
-            assert setup_component(self.hass, notify.DOMAIN, CONFIG)
-        assert handle_config[notify.DOMAIN]
-
-    @patch("os.path.isfile", return_value=True)
-    @patch("os.access", return_value=True)
-    def test_apns_setup_full(self, mock_access, mock_isfile):
-        """Test setup with all data."""
-        config = {
-            "notify": {
-                "platform": "apns",
-                "name": "test_app",
-                "sandbox": "True",
-                "topic": "testapp.appname",
-                "cert_file": "test_app.pem",
-            }
+@pytest.fixture(scope="module", autouse=True)
+def mock_apns_notify_open():
+    """Mock builtins.open for apns.notfiy."""
+    with patch("homeassistant.components.apns.notify.open", mock_open(), create=True):
+        yield
+
+
+@patch("os.path.isfile", Mock(return_value=True))
+@patch("os.access", Mock(return_value=True))
+async def _setup_notify(hass_):
+    assert isinstance(apns.load_yaml_config_file, Mock), "Found unmocked load_yaml"
+
+    with assert_setup_component(1) as handle_config:
+        assert await async_setup_component(hass_, notify.DOMAIN, CONFIG)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)